nixos/modules/services/networking/rosenpass.nix

```diff
@@ -0,0 +1,220 @@
+{ config
+, lib
+, options
+, pkgs
+, ...
+}:
+with builtins;
+with lib; let
+  cfg = config.services.rosenpass;
+  opt = options.services.rosenpass;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.rosenpass = with types; {
+    enable = mkEnableOption "Rosenpass";
+
+    package = mkPackageOption pkgs "rosenpass" { };
+
+    defaultDevice = mkOption {
+      type = nullOr str;
+      description = "Name of the network interface to use for all peers by default.";
+      example = "wg0";
+    };
+
+    settings = mkOption {
+      type = lib.types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          public_key = mkOption {
+            type = path;
+            description = "Path to a file containing the public key of the local Rosenpass peer. Generate this by running {command}`rosenpass gen-keys`.";
+          };
+
+          secret_key = mkOption {
+            type = path;
+            description = "Path to a file containing the secret key of the local Rosenpass peer. Generate this by running {command}`rosenpass gen-keys`.";
+          };
+
+          listen = mkOption {
+            type = listOf str;
+            description = "List of local endpoints to listen for connections.";
+            default = [ ];
+            example = literalExpression "[ \"0.0.0.0:10000\" ]";
+          };
+
+          verbosity = mkOption {
+            type = enum [ "Verbose" "Quiet" ];
+            default = "Quiet";
+            description = "Verbosity of output produced by the service.";
+          };
+
+          peers =
+            let
+              peer = submodule {
+                freeformType = settingsFormat.type;
+
+                options = {
+                  public_key = mkOption {
+                    type = path;
+                    description = "Path to a file containing the public key of the remote Rosenpass peer.";
+                  };
+
+                  endpoint = mkOption {
+                    type = nullOr str;
+                    default = null;
+                    description = "Endpoint of the remote Rosenpass peer.";
+                  };
+
+                  device = mkOption {
+                    type = str;
+                    default = cfg.defaultDevice;
+                    defaultText = literalExpression "config.${opt.defaultDevice}";
+                    description = "Name of the local WireGuard interface to use for this peer.";
+                  };
+
+                  peer = mkOption {
+                    type = str;
+                    description = "WireGuard public key corresponding to the remote Rosenpass peer.";
+                  };
+
+                  extra_params = mkOption {
+                    type = listOf str;
+                    default = [ ];
+                    description = "Extra WireGuard parameters to set. In NixOS, you would probably not want to set this and configure WireGuard via {option}`systemd.network.netdevs`.";
```

Comment:
If it is discouraged we might as well leave it out in the first place.

Reply:
Unresolved this comment for now, since @Majiir is of the opposite opinion.

Reply:
No need - I forgot we had said `extra_params` was not useful in NixOS.